src/focus-ring.js

```diff
@@ -74,11 +80,13 @@ document.addEventListener('DOMContentLoaded', function() {
    * opening a menu or dialog.
    */
   function onKeyDown() {
-    hadKeyboardEvent = true;
     // `activeElement` defaults to document.body if nothing focused,
     // so check the active element is actually focused.
-    if (matches(document.activeElement, ':focus'))
-      addFocusRingClass(document.activeElement);
+    var activeElement = document.activeElement;
+    if (activeElement.tagName == 'BODY')
+      return;
+
+    hadKeyboardEvent = true;
```

Comment:
No worries at all. To answer your question: it's the other way around. The fact the keyboard event precedes the focus event actually works to our advantage here because we know the event sequence is: keydown then focus, and therefore the focus handler can key off of `hadKeyboardEvent` to know focus happened as a result of keyboard, and then add the focus ring class.